buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,49 @@
+plugins {
+    `kotlin-dsl`
+}
+
+gradlePlugin {
+    plugins {
+        register("app") {
+            id = "app"
+            implementationClass = "plugin.ApplicationPlugin"
+        }
+
+        register("library") {
+            id = "library"
+            implementationClass = "plugin.LibraryPlugin"
+        }
+    }
+}
+
+kotlin {
+    sourceSets {
+        named("main") {
+            kotlin.apply {
+                srcDir("buildSrc/src/main/kotlin")
+            }
+        }
+    }
+}
```

Comment:
you don't need this here unless you rename the main source set from java to kotlin